python/lsst/daf/butler/registry/datasets/byDimensions/_storage.py

```diff
@@ -430,3 +402,200 @@ def getDataId(self, id: int) -> DataCoordinate:
             {dimension.name: row[dimension.name] for dimension in self.datasetType.dimensions.required},
             graph=self.datasetType.dimensions
         )
+
+
+class ByDimensionsDatasetRecordStorageInt(ByDimensionsDatasetRecordStorage):
+    """Implementation of ByDimensionsDatasetRecordStorage which uses integer
+    auto-incremented column for dataset IDs.
+    """
+
+    def insert(self, run: RunRecord, dataIds: Iterable[DataCoordinate],
+               idMode: DatasetIdGenEnum = DatasetIdGenEnum.UNIQUE) -> Iterator[DatasetRef]:
```

Comment:
Can we have this raise if `idMode is not DatasetIdGenEnum.UNIQUE`?

Reply:
Will that break when ingest starts using `DatasetIdGenEnum` and someone will want to ingest something into old integers repo?

Reply:
Yes, and I think that's a good thing, at least for now - I want to force the high level code/user to acknowledge what kind of ID they are getting and respond accordingly.

Reply:
I have added an exception if non-UINQUE is passed to `import_` or `insert` for integer IDs.